test/close-pipelining.test.js

```diff
@@ -87,13 +87,16 @@ test('Should close the socket abruptly - pipelining - return503OnClosing: false,
   })
 
   const responses = await Promise.allSettled([
+    instance.request({ path: '/', method: 'GET' }),
     instance.request({ path: '/', method: 'GET' }),
     instance.request({ path: '/', method: 'GET' }),
     instance.request({ path: '/', method: 'GET' })
   ])
+
   t.equal(responses[0].status, 'fulfilled')
-  t.equal(responses[1].status, 'rejected')
+  t.equal(responses[1].status, 'fulfilled')
```

Comment:
It seems that, with the pipelining, and the adjustments to the way we handle the secondary bindings, always at least 2 requests made it through, and from the third one on are rejected.